jerry-debugger/jerry-client-ws.py

```diff
@@ -817,48 +846,59 @@ def release_function(debugger, data):
     logging.debug("Function {0x%x} byte-code released", byte_code_cp)
 
 
-def enable_breakpoint(debugger, breakpoint):
-    if breakpoint.active_index < 0:
-        debugger.next_breakpoint_index += 1
+def enable_breakpoint(debugger, breakpoint, pending):
+    if pending:
+        if not debugger.breakpoint_pending_exists(breakpoint):
+            debugger.next_breakpoint_index += 1
+            breakpoint.index = debugger.next_breakpoint_index
+            debugger.pending_breakpoint_list[debugger.next_breakpoint_index] = breakpoint
+            print("%sPending breakpoint%s at %s" % (debugger.yellow, debugger.nocolor, breakpoint))
+        else:
+            print("%sPending breakpoint%s already exists" % (debugger.yellow, debugger.nocolor))
 
-        debugger.active_breakpoint_list[debugger.next_breakpoint_index] = breakpoint
-        breakpoint.active_index = debugger.next_breakpoint_index
-        debugger.send_breakpoint(breakpoint)
+    else:
+        if breakpoint.active_index < 0:
+            debugger.next_breakpoint_index += 1
+            debugger.active_breakpoint_list[debugger.next_breakpoint_index] = breakpoint
+            breakpoint.active_index = debugger.next_breakpoint_index
+            debugger.send_breakpoint(breakpoint)
 
-    print ("Breakpoint %d at %s" % (breakpoint.active_index, breakpoint))
+        print("%sBreakpoint %d at%s %s" % (debugger.green, breakpoint.active_index, debugger.nocolor, breakpoint))
 
 
-def set_breakpoint(debugger, pending, string):
+def set_breakpoint(debugger, string):
     line = re.match("(.*):(\\d+)$", string)
     found = False
 
     if line:
         source_name = line.group(1)
-        line = int(line.group(2))
+        new_line = int(line.group(2))
 
-        for breakpoint in debugger.line_list.get(line):
+        for breakpoint in debugger.line_list.get(new_line):
             func_source = breakpoint.function.source_name
             if (source_name == func_source or
                     func_source.endswith("/" + source_name) or
                     func_source.endswith("\\" + source_name)):
 
-                enable_breakpoint(debugger, breakpoint)
+                enable_breakpoint(debugger, breakpoint, False)
                 found = True
 
     else:
         for function in debugger.function_list.values():
             if function.name == string:
-                enable_breakpoint(debugger, function.lines[function.first_breakpoint_line])
+                enable_breakpoint(debugger, function.lines[function.first_breakpoint_line], False)
                 found = True
 
     if not found:
-        print("Breakpoint not found")
-        if pending:
+        print("No breakpoint found, do you want to add a pending breakpoint? (y or [n])")
+        ans = sys.stdin.readline()
+        if ans == 'yes\n' or ans == 'y\n':
```

Comment:
A quick note: `ans in ['yes\n', 'y\n']` is more pythonic than `ans == 'yes\n' or ans == 'y\n'`